fork_choice/confirmation-rule.md

```diff
@@ -0,0 +1,673 @@
+        ffg_support_for_checkpoint * 300 + (300 - 3 * confirmation_byzantine_threshold) *
+        remaining_ffg_weight - max_adversarial_ffg_support_for_checkpoint * 300
+    )
+```
+
+#### `is_ffg_confirmed_previous_epoch`
+
+```python
+def is_ffg_confirmed_previous_epoch(
+    total_active_balance: int,
+    ffg_support_for_checkpoint: int,
+    confirmation_byzantine_threshold: int,
+    confirmation_slashing_threshold: int,    
+) -> bool:
+    """
+    Returns whether the `block_root`'s checkpoint is justified
+    """
+
+    max_adversarial_ffg_support_for_checkpoint = int(
+        min(
+            (total_active_balance * confirmation_byzantine_threshold - 1) // 100 + 1,
+            confirmation_slashing_threshold,
+            ffg_support_for_checkpoint
+        )
+    )    
+
+    return 2 * total_active_balance <= (ffg_support_for_checkpoint - max_adversarial_ffg_support_for_checkpoint) * 3
+```
+
+#### `is_ffg_confirmed`
+
+```python
+def is_ffg_confirmed(
+    store: Store,
+    confirmation_byzantine_threshold: int,
+    confirmation_slashing_threshold: int,
+    block_root: Root,
+) -> bool:
+    """
+    Returns whether the `block_root`'s checkpoint is justified
+    """
+    current_epoch = get_current_store_epoch(store)
+
+    block = store.blocks[block_root]
+    block_epoch = compute_epoch_at_slot(block.slot)
+
+    # This function is only applicable to current and previous epoch blocks
+    assert current_epoch <= block_epoch + 1
+    assert block_epoch <= current_epoch
+
+    total_active_balance = int(get_total_active_balance_for_block_root(store, block_root))
+
+    ffg_support_for_checkpoint = int(get_ffg_support(store, block_root))
+
+    if block_epoch == current_epoch:
+        remaining_ffg_weight = int(get_remaining_weight_in_current_epoch(store, block_root))
+        return is_ffg_confirmed_current_epoch(
+            total_active_balance,
+            ffg_support_for_checkpoint,
+            remaining_ffg_weight,
+            confirmation_byzantine_threshold,
+            confirmation_slashing_threshold
+        )
+    else:
+        return is_ffg_confirmed_previous_epoch(
+            total_active_balance,
+            ffg_support_for_checkpoint,
+            confirmation_byzantine_threshold,
+            confirmation_slashing_threshold
+        )  
+```
+
+### `is_confirmed`
+
+```python
+def is_confirmed(
+    store: Store,
+    confirmation_byzantine_threshold: int,
+    confirmation_slashing_threshold: int,
+    block_root: Root
+) -> bool:
+    current_epoch = get_current_store_epoch(store)
+
+    block = store.blocks[block_root]
+    block_state = store.block_states[block_root]
+    block_justified_checkpoint = block_state.current_justified_checkpoint.epoch
+    block_epoch = compute_epoch_at_slot(block.slot)
+
+    # This function is only applicable to current and previous epoch blocks
+    assert current_epoch <= block_epoch + 1
+    assert block_epoch <= current_epoch
```

Comment:
What do you think about the following modification? to make it clearer
```suggestion
    previous_epoch = GENESIS_EPOCH if current_epoch == GENESIS_EPOCH else current_epoch - 1
    assert block_epoch in (previous_epoch, current_epoch)
```

Reply:
That is better than what I had, but I have decided to change this as per @djrtwo's suggestion [here](https://github.com/ethereum/consensus-specs/pull/3339/files#r1255996271) as I think that is even clearer